File: lecture_notes/09 functions/4-rest.js
'use strict';

const catchRest = (...args) => {
  console.log(args);
};

catchRest(1, 2, 3);

const f2 = (...args) => {
  args.forEach((arg) => {
    const type = typeof arg;
    console.log('Type: ' + type);
    if (type === 'object') {
      console.log('Value: ' + JSON.stringify(arg));
    } else {
      console.log('Value: ' + arg);
    }
  });
};

f2(1, 'Marcus', { field: 'value' });

File: lecture_notes/09 functions/4-rest.swift
func catchRest(args: Any...) {
    for val in args {
        print(val)
    }
}

catchRest(args: 1, 2, 3)

func f2(args: Any...) {
    for val in args {
        print("Type: ", type(of: val))
        print("Value:", val)
    }
}

f2(args: 1, "Marcus", "Value")
